### DISCUSSION TITLE
Button tapgesture no longer working in MAUI 8 iOS

### DISCUSSION THREAD

**Discussion by Ghostbird** [dotnet/maui#19083] [Q&A]
Are we doing something wrong? We upgraded our app from MAUI 7 to MAUI 8. On Android it went smoothly, but on iOS every button whose click handling relied on `TapGestureRecognizer` stopped working. Since we dynamically load our UI, this meant that almost everything in our app stopped working, except the static settings screen, login and logout.

Since I can't fathom that the people behind MAUI would release a new version where something as basic as buttons don't work, I'm really puzzled what we're doing wrong.

Anything that adds a `TapGestureRecogniser` to a `Button` seems to break the button functionality, including `Behavior`s for animation and haptic feedback.

**Comment by sjordanGSS** [comment]
Why are you adding a ```TapGestureRecognizer``` to a ```Button```? ```Button``` has its own properties for ```Command```, ```CommandParameter``` and ```Clicked```, so you don't need to add a ```TapGestureRecognizer``` to receive user interaction with them.

**Comment by Ghostbird** [comment]
Because it's the obvious API to use that's identical across multiple
dynamically added controls.

But that's kind of off-topic, the question is. What suddenly caused it to
stop working, just on iOS?

**Reply by Stedy59** [reply]
Not off-topic... Since you are using a base control that supports "TapGestureRecognizer" interactions, I would suggest not using a button as your base, or at least setting the input "transparency".

**Reply by Ghostbird** [reply]
I don't really get what you're trying to say here, but I've upgraded this to #19099.
